### ISSUE TITLE
Error accessing gzipped file

### ISSUE THREAD

**Issue by Onager** [log2timeline/dfvfs#242]
Exception raised from Plaso, but this is a dfvfs error:
```
2017-10-23 17:18:51,335 [ERROR] (Worker_02 ) PID:17385 <worker_process> Unable to decompress zlib compressed stream with error: Error -3 while decompressing: invalid stored block lengths.
Traceback (most recent call last):
  File "/Users/onager/code/onager_plaso/plaso/multi_processing/worker_process.py", line 210, in _ProcessPathSpec
    extraction_worker.ProcessPathSpec(parser_mediator, path_spec)
  File "/Users/onager/code/onager_plaso/plaso/engine/worker.py", line 777, in ProcessPathSpec
    self._ProcessFileEntry(mediator, file_entry)
  File "/Users/onager/code/onager_plaso/plaso/engine/worker.py", line 596, in _ProcessFileEntry
    mediator, file_entry, data_stream.name)
  File "/Users/onager/code/onager_plaso/plaso/engine/worker.py", line 641, in _ProcessFileEntryDataStream
    self._AnalyzeDataStream(mediator, file_entry, data_stream_name)
  File "/Users/onager/code/onager_plaso/plaso/engine/worker.py", line 140, in _AnalyzeDataStream
    self._AnalyzeFileObject(mediator, file_object)
  File "/Users/onager/code/onager_plaso/plaso/engine/worker.py", line 177, in _AnalyzeFileObject
    data = file_object.read(maximum_read_size)
  File "/Users/onager/code/virtual_envs/pycharm_onager_plaso/lib/python2.7/site-packages/dfvfs/file_io/file_object_io.py", line 99, in read
    return self._file_object.read(size=size)
  File "/Users/onager/code/virtual_envs/pycharm_onager_plaso/lib/python2.7/site-packages/dfvfs/file_io/compressed_stream_io.py", line 220, in read
    self._AlignUncompressedDataOffset(self._current_offset)
  File "/Users/onager/code/virtual_envs/pycharm_onager_plaso/lib/python2.7/site-packages/dfvfs/file_io/compressed_stream_io.py", line 149, in _AlignUncompressedDataOffset
    read_count = self._ReadCompressedData(self._COMPRESSED_DATA_BUFFER_SIZE)
  File "/Users/onager/code/virtual_envs/pycharm_onager_plaso/lib/python2.7/site-packages/dfvfs/file_io/compressed_stream_io.py", line 177, in _ReadCompressedData
    self._decompressor.Decompress(self._compressed_data))
  File "/Users/onager/code/virtual_envs/pycharm_onager_plaso/lib/python2.7/site-packages/dfvfs/compression/zlib_decompressor.py", line 50, in Decompress
    '{0!s}.').format(exception))
BackEndError: Unable to decompress zlib compressed stream with error: Error -3 while decompressing: invalid stored block lengths.
```

But the file is readable with gunzip, as well as the python gzip module:
 ```
import gzip
path  = '/path/to/file'
g = gzip.open(path, 'r')
decompressed = g.read()
print g
```

**Comment by Onager** [comment]
Sample file: https://github.com/Onager/plaso/blob/fsevents/test_data/fsevents-0000000002d89b58

**Comment by Onager** [comment]
That sample file doesn't actually work, https://github.com/Onager/dfvfs/blob/gzip/test_data/fsevents_000000000000b208 triggers this issue

**Comment by Onager** [comment]
After a closer look, this is because the gzip implementation in dfvfs only supports gzip files with a single member (see http://www.zlib.org/rfc-gzip.html). _ReadFileFooter() just reads the last 8 bytes to determine the "file footer": https://github.com/log2timeline/dfvfs/blob/master/dfvfs/file_io/gzip_file_io.py#L133 but in reality, this is the trailer of the last compressed member in the file, not a true file footer.

**Comment by Onager** [comment]
Fixed in #257
